docs/DESIGN.md

```diff
@@ -1,95 +1,187 @@
+
+![Operator Reconciler Architecture](images/odh-operator-reconciliation.png)
 
-### DataScienceCluster
+### User-facing Custom Resources and their controllers
 
-- This CR will be watched by the ODH operator to enable various data science components.
-- It is responsible for enabling support for CRDs like Notebooks, DataSciencePipelinesApplication, InferenceService etc. based on the configuration
-- Initially only one instance of DataScienceCluster CR will be supported by the operator. A user can extend/update the CR to enable/disable components.
-- Detailed API fields are described in the CRD.
+The two user-facing CRs are **DSCInitialization** (DSCI) and **DataScienceCluster** (DSC).
+Together, these CRs are responsible for configuration of ODH at the platform level.
+
+#### DSCInitialization (DSCI)
+
+- DSCI CR is created and managed by the ODH operator to perform initial setup that is common for all components.
+  - for example, DSCI defines the namespace for ODH component deployments, monitoring config, ServiceMesh config,...
+- Some examples of initial setup include creating namespaces, network policies, SCCs, common configmaps and secrets.
+- DSCI instance is a singleton in the k8s/OCP cluster, i.e., only one instance of this CR will always be present in the cluster.
+- DSCI can be deleted to re-run the initial setup without requiring rebuild of the operator.
+- DSCI controller implementation can be found in `internal/controller/dscinitialization` directory.
+
+#### DataScienceCluster (DSC)
+
+- DSC CR is watched by the ODH operator and serves as a single point to configure and enable various ODH components.
+- It is responsible for enabling support for CRDs like Dashboard, Workbenches, Data Science Pipelines, etc.
+- Like with DSCI, DSC instance is also a singleton in the k8s/OCP cluster. A user can extend/update the CR to enable/disable components.
+- DSC controller implementation can be found in `internal/controller/datasciencecluster` directory.
+- Detailed API fields are described in the CRD. Example DSC configurations are provided in the [Examples section](#examples).
+
+### Component CRs and controllers
+
+The ODH operator reconciliation refactor introduced dedicated CRDs for each ODH component.
+From there, if enabled, an ODH component is represented by its dedicated singleton CR instance and the owned subresources, all of which are managed by a dedicated component controller. Component CRs are owned by the DSC.
+The component controller implementations are located in the subdirectories of `internal/controller/components` directory.
+ 
+List of currently integrated ODH components:
+
+| **ODH Component** | **CR Kind** |
+| --- | --- |
+| CodeFlare | `CodeFlare` |
+| ODH Dashboard | `Dashboard` |
+| Data Science Pipelines | `DataSciencePipelines` |
+| Feature Store (Feast) | `FeastOperator` |
+| KServe | `Kserve` |
+| Kueue | `Kueue` |
+| Model Controller | `ModelController` |
```

Comment:
this should not be listed

Reply:
The original document contained the list of supported components. Is this about removing the CR Kind column, or removing the whole table?

Reply:
only this line "Model Controller " which is for user to interact